### PR TITLE
fix: add cli validate and name check

### PR DESCRIPTION
**Pull request by xiaolang124**
1. Add cli validate env parameter.
2. Add cli provision new env name check.

image

### CODE REVIEW

File: packages/cli/src/helpParamGenerator.ts [hunk 1]
```diff
@@ -204,6 +204,8 @@ export class HelpParamGenerator {
         nodes = nodes.concat([EnvNode]);
       } else if (HelpParamGenerator.showEnvStage.indexOf(stage) >= 0) {
         nodes = nodes.concat([EnvNodeNoCreate]);
+      } else if (stage == "validate") {
```
Comment (qinezh) [inline packages/cli/src/helpParamGenerator.ts:207]: move `validate` to the list of `showEnvStage`?

File: packages/fx-core/src/core/error.ts [hunk 2]
```diff
@@ -158,11 +158,20 @@ export function ProjectSettingsUndefinedError(): FxError {
...
+export function ProjectEnvNotExistError(env: string): FxError {
   return new UserError(
     "ProjectEnvNotExistError",
     `The specified env ${env} does not exist. Select an existing env.`,
     CoreSource,
     new Error().stack
   );
 }
+
+export function ProjectEnvIllegalError(): FxError {
```
Comment (qinezh) [inline packages/fx-core/src/core/error.ts:170]: How about InvalidEnvNameError?

File: packages/cli/src/helpParamGenerator.ts (hunk 1 above)
Comment (xiaolang124) [reply packages/cli/src/helpParamGenerator.ts:207]: Fixed.

File: packages/fx-core/src/core/error.ts (hunk 2 above)
Comment (xiaolang124) [reply packages/fx-core/src/core/error.ts]: Fixed.

**Review by qinezh — APPROVED** [review]

File: packages/vscode-extension/src/utils/commonUtils.ts [hunk 3]
```diff
@@ -105,7 +105,8 @@ export function anonymizeFilePaths(stack?: string): string {
   }
 
   const nodeModulesRegex = /^[\\\/]?(node_modules|node_modules\.asar)[\\\/]/;
-  const fileRegex = /(file:\/\/)?([a-zA-Z]:(\\\\|\\|\/)|(\\\\|\\|\/))?([\w-\._]+(\\\\|\\|\/))+[\w-\._]*/g;
+  const fileRegex =
+    /(file:\/\/)?([a-zA-Z]:(\\\\|\\|\/)|(\\\\|\\|\/))?([\w-\._]+(\\\\|\\|\/))+[\w-\._]*/g;
```
Comment (LongOddCode) [inline packages/vscode-extension/src/utils/commonUtils.ts:109]: shall we add some ut for this regex?

**Review by LongOddCode — APPROVED** [review]

File: packages/vscode-extension/src/utils/commonUtils.ts (hunk 3 above)
Comment (xiaolang124) [reply packages/vscode-extension/src/utils/commonUtils.ts:109]: This is telemetry old code. I think this will be added when necessary.

**Review by Alive-Fish — APPROVED** [review]
lgtm

**Review by KennethBWSong — APPROVED** [review]

**Pull request merged**